infra/platforms/improve.py

```diff
@@ -168,15 +173,32 @@ def main():
         clang_results = None
 
     # run Herbie for output cores
-    cores = runner.herbie_improve(cores=input_cores, threads=herbie_threads)
+    cores = runner.herbie_improve(cores=input_cores, threads=herbie_threads, localize=True, old_cost=False)
 
-    # analyze input and output cores
     analyze_cores(runner, input_cores + cores)
     run_cores(runner, input_cores, py_sample)
     driver_dirs = run_cores(runner, cores, py_sample)
 
-    # publish results
-    runner.write_improve_report(all_input_cores, cores, driver_dirs, clang_results)
+    if ablation:
+        ablation_cores = []
+        ablation_times = []
+        ablation_frontiers = []
+        for ablation in ablation_map:
+            localize, cost = ablation_map[ablation]
+
+            # run Herbie improve and get associated sampled points
+            cores = runner.herbie_improve(cores=input_cores, threads=herbie_threads, localize=localize, old_cost=cost)
+            analyze_cores(runner, input_cores + cores)
+            run_cores(runner, input_cores, py_sample)
+            driver_dirs = run_cores(runner, cores, py_sample)
```

Comment:
You're overwriting `cores` and `driver_dirs` in the loop. Each treatments should produce a pair of `(cores, driver_dirs)` which should then be written to the JSON.